src/app/accounts/set-username.ts

```diff
@@ -16,8 +16,7 @@ export const setUsername = async ({
   const accountsRepo = AccountsRepository()
   const account = await accountsRepo.findById(id as AccountId)
   if (account instanceof Error) return account
-
-  if (account.username !== undefined) return new UsernameIsImmutableError()
+  if (account.username != null) return new UsernameIsImmutableError()
```

Comment:
did you test this with current data? maybe is better to validate undefined, null and empty

```suggestion
  if (account.username) return new UsernameIsImmutableError()
```

Reply:
it's been write as null in mongodb. maybe there are some scenario where it is undefined. it can't be empty string IMU because there is a non empty index on it.

Reply:
if it is a possibility then we should use `if (account.username)`